modules/tinymce/src/core/main/ts/fmt/RemoveFormat.ts

```diff
@@ -210,16 +210,11 @@ const removeFormat = (ed: Editor, format: RemoveFormatPartial, vars?: FormatVars
   // Applies to styling elements like strong, em, i, u, etc. so that if they have styling attributes, the attributes can be kept but the styling element is removed
   if (format.inline && format.remove === 'all' && Type.isArray(format.preserve_attributes)) {
     // Remove all attributes except for the attributes specified in preserve_attributes
-    Arr.each(dom.getAttribs(elm), (attr) => {
-      if (attr) {
-        const attrName = attr.nodeName.toLowerCase();
-        if (!Arr.exists(format.preserve_attributes, (pAttr) => pAttr === attrName)) {
-          dom.setAttrib(elm, attrName, '');
-        }
-      }
-    });
+    const attrsToPreserve = Arr.filter(dom.getAttribs(elm), (attr) => Arr.contains(format.preserve_attributes, attr.name.toLowerCase()));
+    dom.removeAllAttribs(elm);
+    Arr.each(attrsToPreserve, (attr) => dom.setAttrib(elm, attr.name, attr.value));
```

Comment:
I'm surprised we're not doing the opposite and removing only attrs that aren't in the setting?

Reply:
I think for peace of mind I prefer that all of the attributes are removed first before applying the preserved attributes. In theory, removing the attributes that are not contained in `preserve_attributes` should work exactly the same but I have found some flakiness with that approach (though it wasn't the exact same code).